src/main/scala/is/hail/table/Table.scala

```diff
@@ -1056,34 +1085,33 @@ class Table(val hc: HailContext, val ir: TableIR) {
 
   def collect(): IndexedSeq[Row] = rdd.collect()
 
-  def write(path: String, overwrite: Boolean = false) {
-    if (!path.endsWith(".kt") && !path.endsWith(".kt/"))
-      fatal(s"write path must end in '.kt', but found '$path'")
+  def write(path: String, overwrite: Boolean = false, codecSpecJSONStr: String = null) {
```

Comment:
This isn't a performance critical place, we generally have preferred explicit option types over relying on the nullability of reference types.

Reply:
I generally have a preference away from Option, period.  py4j is already known to be painfully slow and changing this to an option just adds more round trips and I think all the option nonsense just leads to noisier and more error prone Python.  If you push, I will change.